lua/keymappings.lua

```diff
@@ -1,106 +1,115 @@
-    { "<C-l>", "<C-w>l", { silent = true } },
+    { "<C-h>", "<C-w>h" },
+    { "<C-j>", "<C-w>j" },
+    { "<C-k>", "<C-w>k" },
+    { "<C-l>", "<C-w>l" },
 
     -- Resize with arrows
-    { "<C-Up>", ":resize -2<CR>", { silent = true } },
-    { "<C-Down>", ":resize +2<CR>", { silent = true } },
-    { "<C-Left>", ":vertical resize -2<CR>", { silent = true } },
-    { "<C-Right>", ":vertical resize +2<CR>", { silent = true } },
+    { "<C-Up>", ":resize -2<CR>" },
+    { "<C-Down>", ":resize +2<CR>" },
+    { "<C-Left>", ":vertical resize -2<CR>" },
+    { "<C-Right>", ":vertical resize +2<CR>" },
 
     -- Tab switch buffer
     -- { "<TAB>", ":bnext<CR>" },
     -- { "<S-TAB>", ":bprevious<CR>" },
 
-    -- Move current line / block with Alt-j/k a la vscode.
+    -- keys.ve current line / block with Alt-j/k a la vscode.
     { "<A-j>", ":m .+1<CR>==" },
     { "<A-k>", ":m .-2<CR>==" },
 
     -- QuickFix
     { "]q", ":cnext<CR>" },
     { "[q", ":cprev<CR>" },
+    { "<C-q>", ":call QuickFixToggle()<CR>" },
 
     -- {'<C-TAB>', 'compe#complete()', {noremap = true, silent = true, expr = true}},
   },
-  t = { -- Terminal mode
+
+  term_mode = { -- Terminal mode
     -- Terminal window navigation
     { "<C-h>", "<C-\\><C-N><C-w>h" },
     { "<C-j>", "<C-\\><C-N><C-w>j" },
     { "<C-k>", "<C-\\><C-N><C-w>k" },
     { "<C-l>", "<C-\\><C-N><C-w>l" },
   },
-  v = { -- Visual/Select mode
+
+  visual_mode = { -- Visual/Select mode
     -- Better indenting
     { "<", "<gv" },
     { ">", ">gv" },
 
     -- { "p", '"0p', { silent = true } },
     -- { "P", '"0P', { silent = true } },
   },
-  x = { -- Visual mode
-    -- Move selected line / block of text in visual mode
+
+  visual_block_mode = { -- Visual mode
+    -- keys.ve selected line / block of text in visual mode
     { "K", ":move '<-2<CR>gv-gv" },
     { "J", ":move '>+1<CR>gv-gv" },
 
-    -- Move current line / block with Alt-j/k ala vscode.
+    -- keys.ve current line / block with Alt-j/k ala vscode.
     { "<A-j>", ":m '>+1<CR>gv-gv" },
     { "<A-k>", ":m '<-2<CR>gv-gv" },
   },
-  [""] = {
-    -- Toggle the QuickFix window
-    { "<C-q>", ":call QuickFixToggle()<CR>" },
-  },
 }
 
--- TODO: fix this
 if vim.fn.has "mac" == 1 then
-  mappings["n"][5][1] = "<A-Up>"
-  mappings["n"][6][1] = "<A-Down>"
-  mappings["n"][7][1] = "<A-Left>"
-  mappings["n"][8][1] = "<A-Right>"
+  -- TODO: fix this
+  default_keys.insert_mode["n"][5][1] = "<A-Up>"
+  default_keys.insert_mode["n"][6][1] = "<A-Down>"
+  default_keys.insert_mode["n"][7][1] = "<A-Left>"
+  default_keys.insert_mode["n"][8][1] = "<A-Right>"
 end
 
-register_mappings(mappings, { silent = true, noremap = true })
+if O.keys.leader_key == " " or O.keys.leader_key == "space" then
+  vim.g.mapleader = " "
+else
+  vim.g.mapleader = O.keys.leader_key
+end
 
-vim.cmd 'inoremap <expr> <c-j> ("\\<C-n>")'
-vim.cmd 'inoremap <expr> <c-k> ("\\<C-p>")'
+local function get_user_provided(mode)
```

Comment:
Are you ok with renaming this to `get_user_keys` ?

Reply:
something something hardest part of programming is naming 😅